lib/ops/opsservice/site.go

```diff
@@ -604,6 +605,49 @@ func (s *site) getClusterEnvironmentVariables() (env storage.EnvironmentVariable
 	return env, nil
 }
 
+// updateServiceConfiguration updates the gravity-site service configuration if
+// the gravityControllerService configuration has been modified.
+func (s *site) updateServiceConfiguration() error {
+	resource, err := s.getClusterConfiguration()
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	controllerConfig := resource.GetGravityControllerServiceConfig()
+	if controllerConfig.IsEmpty() {
+		return nil
+	}
+
+	client, err := s.service.GetKubeClient()
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	services := client.CoreV1().Services(defaults.KubeSystemNamespace)
+
+	svc, err := services.Get("gravity-site", metav1.GetOptions{})
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	svc.Spec.Type = v1.ServiceType(controllerConfig.Type)
```

Comment:
I think this should check whether the service already has correct type and annotations and do nothing in this case.